core/services/relay/delegate/delegate.go

```diff
@@ -0,0 +1,144 @@
+package delegate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	"github.com/smartcontractkit/chainlink/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/pkg/errors"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/relay"
+	"github.com/smartcontractkit/chainlink/core/services/relay/ethereum"
+	"go.uber.org/multierr"
+)
+
+var (
+	_ relay.Relayer = &ethereum.Relayer{}
+	_ relay.Relayer = &solana.Relayer{}
+)
+
+type delegate struct {
+	relayers map[relay.Network]relay.Relayer
+	ks       keystore.Master
+}
+
+func NewRelayDelegate(db *sqlx.DB, ks keystore.Master, chainSet evm.ChainSet, lggr logger.Logger) *delegate {
+	return &delegate{
+		ks: ks,
+		relayers: map[relay.Network]relay.Relayer{
+			relay.Ethereum: ethereum.NewRelayer(db, chainSet, lggr),
+			relay.Solana:   solana.NewRelayer(lggr),
+		},
+	}
+}
+
+// A delegate relayer on start will start all relayers it manages.
+func (d delegate) Start() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Start())
```

Comment:
hmm... this feels wrong.

We would end up starting ethereum and solana relayers even if there are no ethereum or solana jobs.

Have we made an architectural mistake?

Reply:
right now their relayer Start methods do nothing, but we do need the Close for clean if there are jobs so it just uses the Service interface. Seems ok to leave as is, but we could only have the Close method if you like

Reply:
A relay gets an `onStart` hook (before any job). It can use it to start a head-tracker for example, if the relay is configured for just one network. Kinda how Ethereum relay could work, currently the head tracker is started outside by `evm.Chain` component.

The Solana relay currently does nothing on start but starts running internal processes just as first job is configured.